### DISCUSSION TITLE
can't get essential reactions for tasks , biomass not protected

### DISCUSSION THREAD

**Discussion by PkiwiBird** [SysBioChalmers/Human-GEM#404] [Q&A] [answered]
Hi,
I ran the  `gen_all_tINIT_model.m` script exactly like mentioned in Human1 article (with same package versions), but with version 12 of Human-GEM instead of the old one. 
1. in the old version of human1 the 'unconstrained' field had 1's in all 'x' compartment metabolites (which were boundary at the time), but version 12 doesn't have boundary metabolites (only 'e' metabolites). Does it mean the unconstrained field in version 12 should be filled only with 0's or does it need to have '1s' at the 'e' metabolites?
2. Although all tasks pass in `checkTasks` (I replaced '[s]' by '[e]' in tasks file)  there are no essential reactions for biomass task. the `essentialRxnMat` is filled with 0's (sum(essentialRxnMat(:,57)) is 0). Do you know what might be the cause?

**Comment by JonathanRob** [comment, marked as answer]
Hi @PkiwiBird, I wouldn't recommend using the `gen_all_tINIT_model.m` with newer functions/models - it is not maintained and therefore not likely to be compatible with new model or package versions (as you are experiencing). I will anyway try to address your questions, but I cannot promise that we'll be able to fix it easily.

1. The new Human-GEM model uses `b` to denote boundary metabolites. So the unconstrained field should contain a value of `1` for metabolites in the `b` compartment.
2. What version of the essential reactions metabolic task file are you using? Also, make sure that boundary metabolites are added to the model prior to running `checkTasks` (i.e., using the `addBoundaryMets` function).

**Reply by PkiwiBird** [reply]
thank you. addBoundaryMets made it work. is there a way to directly protect a reaction that is not essential for a task?

**Reply by JonathanRob** [reply]
By "protect a reaction" I assume you mean to require that a specific reaction(s) from the reference model is included in the extracted tINIT model - which is a good question. Unfortunately, there is not really a straightforward way to do this, but it is possible.

If the reaction you want to keep is the only reaction that produces a certain metabolite, then you can add a new task to the metabolic task document requiring that this metabolite is produced with a lower flux bound of 1, for example. This is a pretty specific case though, and it is not likely that the reaction(s) you want to keep will meet this criteria. If you can otherwise define a metabolic task that is specific enough that your reaction(s) of interest is absolutely necessary to pass that task, this will be the best way to make sure that the reaction will end up in your tINIT model, and that it will be able to carry flux.

Another option is to just add the reaction back to the extracted model after running tINIT. The drawback of this approach is that it is not guaranteed that the reaction will be able to carry flux when it is added to the model. But one can look at which reactions/metabolites are necessary for the reaction to carry flux (e.g., by maximizing or minimizing flux through the target reaction in the full model, and looking at which reactions carry flux), and adding those to the tINIT model.

**Reply by PkiwiBird** [reply]
Ok. Or to find the necessary reactions to the reaction of interest, one could maybe set the LB of that reaction to a positive value in the generic model and then minimize an objective that is the sum of all fluxes in the model (to find the minimum flux distribution that allows flux in that reaction). The reactions carrying flux in that situation could be considered necessary.
Thank you for the complete explanation.

**Reply by JonathanRob** [reply]
Yes, exactly, that should also work.
